src/CatSearch.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react"
+import { useState, useEffect, useCallback } from "react"
+import { fetchSearchResultsFromAPI } from "./searchApiClient"
+
+export function CatSearch() {
+  const [valid, setValid] = useState(false)
+  const [hits, setHits] = useState<any[]>([])
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const searchTermOnChange = useCallback((event) => {
+    const { value } = event.target
+    setSearchTerm(value)
+  }, [])
+
+  useEffect(() => {
+    setValid(true)
+    if (searchTerm.length < 2) {
+      setValid(false)
+      return
+    }
+
+    fetchSearchResultsFromAPI(searchTerm).then((hits) => {
+      setHits(hits)
+    })
+  })
+
+  return (
+    <div>
+      <h2>Search for cat breed</h2>
+      <input type="text" onChange={searchTermOnChange} />
+      <div>
+        {valid === false
```

Comment:
To make this code a little more readable, you can change this assertion to `valid` and flip the order of the strings. So, "You search for..." would come after the `?`.